lib/data/external/GcpClient.js

```diff
@@ -97,6 +123,70 @@ class GcpClient extends AwsClient {
         });
     }
 
+    put(stream, size, keyContext, reqUids, callback) {
```

Comment:
If the `AwsClient.put()` implementation can be considered the default implementation for backends (same for `_createAwsKey`), it seems you could just set `this._supportsVersioning = true` and keep using it to avoid code duplication, but I may miss something.